SOURCE/Main_DOODZ.c

```diff
@@ -763,20 +764,26 @@ int main( int nargs, char *args[] ) {
             // If Picard 2 Newton is activated: force initial Newton step
             if ( Newton == 1 && Nmodel.Picard2Newton == 1 ) {
                 model.Newton = 0;
-                first_Newton = 1;
+                FirstNewton  = 1;
             }
             
             while ( Nmodel.nit <= Nmax_picard && nstag<model.nstagmax) {
                 
                 if ( Nmodel.nit > 0 && Nmodel.Picard2Newton == 1 ) {
                     if ( rx_rel[Nmodel.nit-1] < Nmodel.Pic2NewtCond || rz_rel[Nmodel.nit-1] < Nmodel.Pic2NewtCond || rp_rel[Nmodel.nit-1] < Nmodel.Pic2NewtCond || Nmodel.nit>= Nmodel.nit_Pic_max) {
-                        if ( first_Newton == 1 ) cholmod_free_factor ( &CholmodSolver.Lfact, &CholmodSolver.c);
-                        if ( first_Newton == 1 ) CholmodSolver.Analyze = 1;
-                        if ( first_Newton == 0 ) CholmodSolver.Analyze = 0;
-                        if ( first_Newton == 1 ) first_Newton = 0;
+                        if ( FirstNewton == 1 ) cholmod_free_factor ( &CholmodSolver.Lfact, &CholmodSolver.c);
+                        if ( FirstNewton == 1 ) CholmodSolver.Analyze = 1;
+                        if ( FirstNewton == 0 ) CholmodSolver.Analyze = 0;
+                        if ( FirstNewton == 1 ) FirstNewton = 0;
                         model.Newton          = 1;
                     }
                 }
+
+                // Determine wheter Jacobian matrix should be assembled
+                if ( model.Newton == 1 || model.aniso == 1 ) {
```

Comment:
And also it is unclear to which property `*Newt_on` points to and what's the difference between it and above mentioned properties

Reply:
This should also be addressed. I have not changed the switched model.Newton to model.IsNewton (although we should) because it is present in other .c files. We would need a script to do that.